### PR TITLE
Fixes NGMX-421 - Updates readme

### PR DESCRIPTION
**Pull request by cconvey**
## Description ##
*Quick, minimal* changes to the `NGRPAH_README.md` file in preparation for our open-source release.

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by cconvey** [comment]
@mbrookhart : I've tried to address your first-round feedback.  I had to guess about a few details:
- What exactly do we want to tell users about the values for `OMP_NUM_THREADS` and `KMP_AFFINITY`?  I don't think we want to replicate a tuning guide for those variables.  But I also don't know how much we can deviate from those values to avoid having `train_cifar10.py` crash.
- I fixed up (I think) the paths to the three examples that we claim work.  Some of the paths previously mentioned weren't quite right.

Also, I'm still waiting to hear from you and/or @adstraw about the way the build system handles automated downloading/building of ngraph.  The docs are written based on the current behavior, but that might change.

File: NGRAPH_README.md [hunk 1]
```diff
@@ -1,71 +1,212 @@
...
+       python-pip \
+       python-scipy \
+       python-sklearn \
+       python3-pip \
+       virtualenv
+     ```
+1. **(Optional) Build and install the Intel® nGraph™ open source C++ library and
+   compiler**
+
+   The ngraph library is a build-time and runtime dependency of the ngraph-enabled
```
Comment (indie) [inline NGRAPH_README.md]: nGraph

File: NGRAPH_README.md [hunk 2]
```diff
@@ -1,71 +1,212 @@
...
+   MXNet software.  The build system for ngraph-enabled MXNet supports two approaches
+   to satisfying this dependency:
+
+   - The build system can download and install the ngraph library on its own, as part
+     of the ngraph-enabled MXNet build process.
+
+   - The user can manually install the ngraph library onto the system before building
```
Comment (indie) [inline NGRAPH_README.md]: Better link for this section:  http://ngraph.nervanasys.com/docs/latest/framework-integration-guides.html#mxnet-intg

File: NGRAPH_README.md [hunk 3]
```diff
@@ -1,71 +1,212 @@
...
+     ngraph-enabled MXNet.  To install the ngraph software manually, please see
+     [this page](https://github.com/NervanaSystems/ngraph) for instructions.
+
+   Additional details are provided below.
+1. **Build the ngraph-MXNet libraries**
+
+   The build system for ngraph-enabled MXNet use Gnu Makefiles and is performed
```
Comment (indie) [inline NGRAPH_README.md]: GNU

File: NGRAPH_README.md [hunk 4]
```diff
@@ -1,71 +1,212 @@
...
+   in the source directory.
+
+   The simplest invocation of the build system is as follows:
+
+     ``` sh
+     cd MXNET_ROOT
+     make
+     ```
+
+   Successful completion of the build process results in the creation of the
```
Comment (indie) [inline NGRAPH_README.md]: "Completion of the build process results in the successful creation of the files ... "

File: NGRAPH_README.md (hunk 3 above)
Comment (cconvey) [reply NGRAPH_README.md]: Who, nice nerd-catch!
